cfme/fixtures/service_fixtures.py

```diff
@@ -35,6 +36,25 @@ def catalog_item(appliance, provider, provisioning, dialog, catalog):
     return catalog_item
 
 
+@pytest.fixture(scope="module")
+def generic_catalog_item(request, appliance, dialog, catalog):
+    cat = _service_templates(
+        request, appliance, service_dialog=dialog, service_catalog=catalog, num=1
+    )[0]
+
+    yield appliance.collections.catalog_items.instantiate(
+        appliance.collections.catalog_items.GENERIC,
+        name=cat.name,
+        description=cat.description,
+        display_in=True,
+        catalog=catalog,
+        dialog=dialog,
+    )
+
+    if cat.exists:
```

Comment:
You should be able to use `cat.delete_if_exists`, as that method is defined at the `BaseEntity` level now.

Reply:
nvm, I see you have to call through `cat.action.delete()`